package.json

```diff
@@ -38,13 +38,18 @@
     }
   },
   "dependencies": {
-    "revery": "revery-ui/revery#431e515a00ee46f7bc614a10c2a219aae13179e0",
-    "@opam/dune": "2.6.0",
+    "revery": "*",
+    "@opam/dune": "^2.6.0",
     "@revery/timber": "*",
     "esy-macdylibbundler": "*"
   },
+  "resolutions": {
+    "revery": "revery-ui/revery#d737fcd",
+    "esy-skia": "revery-ui/esy-skia#6106223",
+    "@esy-ocaml/reason": "EduardoRFS/reason:reason.json#35aa4df3de0daa60bdc1133dcf97855decac48f7"
+  },
   "devDependencies": {
-    "ocaml": "~4.9.0",
-    "@opam/ocaml-lsp-server": "ocaml/ocaml-lsp:ocaml-lsp-server.opam#cb8aba2238ba5b0e5a6fe0a3180f25077496f75a"
+    "ocaml": "~4.12",
```

Comment:
Need to update ocaml to 4.12 or we get the following build error

<details>

```
info building revery@github:revery-ui/revery#d737fcd@d41d8cd9
error: build failed with exit code: 1
  build log:
    # esy-build-package: building: revery@github:revery-ui/revery#d737fcd
    # esy-build-package: pwd: /Users/ragnarhardarson/.esy/source/i/revery__c93689c4
    # esy-build-package: running: 'dune' 'build' '-p' 'reason-harfbuzz,reason-skia,reason-sdl2,Revery'
        discover packages/reason-sdl2/src/c_flags.sexp,packages/reason-sdl2/src/c_library_flags.sexp,packages/reason-sdl2/src/cxx_flags.sexp,packages/reason-sdl2/src/flags.sexp
    SDL2 Library Folder Path: /Users/ragnarhardarson/.esy/3________________________________________________________/i/esy_sdl2-2.0.14000-3a4931c9/lib
             gcc packages/reason-skia/src/wrapped/c/c_stubs.o
    c_stubs.c:30:53: warning: incompatible function pointer types passing 'void *(void *, const char *)' to parameter of type 'gr_gl_get_proc' (aka 'void (*(*)(void *, const char *))(void)') [-Wincompatible-function-pointer-types]
                gr_glinterface_assemble_gl_interface(0, reason_skia_sdl2_get);
                                                        ^~~~~~~~~~~~~~~~~~~~
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/esy_skia-2abe672c/include/c/gr_context.h:40:97: note: passing argument to parameter 'get' here
    SK_C_API const gr_glinterface_t* gr_glinterface_assemble_gl_interface(void* ctx, gr_gl_get_proc get);
                                                                                                    ^
    c_stubs.c:29:23: warning: initializing 'gr_glinterface_t *' (aka 'struct gr_glinterface_t *') with an expression of type 'const gr_glinterface_t *' (aka 'const struct gr_glinterface_t *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        gr_glinterface_t *interface =
                          ^
    c_stubs.c:36:55: warning: incompatible function pointer types passing 'void *(void *, const char *)' to parameter of type 'gr_gl_get_proc' (aka 'void (*(*)(void *, const char *))(void)') [-Wincompatible-function-pointer-types]
                gr_glinterface_assemble_gles_interface(0, reason_skia_sdl2_get);
                                                          ^~~~~~~~~~~~~~~~~~~~
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/esy_skia-2abe672c/include/c/gr_context.h:41:99: note: passing argument to parameter 'get' here
    SK_C_API const gr_glinterface_t* gr_glinterface_assemble_gles_interface(void* ctx, gr_gl_get_proc get);
                                                                                                      ^
    c_stubs.c:35:23: warning: initializing 'gr_glinterface_t *' (aka 'struct gr_glinterface_t *') with an expression of type 'const gr_glinterface_t *' (aka 'const struct gr_glinterface_t *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        gr_glinterface_t *interface =
                          ^
    4 warnings generated.
      ocamlmklib src/Native/libRevery_Native_stubs.a
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(ReveryGtk.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(ReveryGtk_Widget.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(dialog_gtk.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(dialog_win32.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(environment_linux.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(environment_windows.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(icon_win32.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(locale_win32.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(shell_gtk.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(shell_win32.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(ReveryGtk.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(ReveryGtk_Widget.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(dialog_gtk.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(dialog_win32.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(environment_linux.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(environment_windows.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(icon_win32.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(locale_win32.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(shell_gtk.o) has no symbols
    /Library/Developer/CommandLineTools/usr/bin/ranlib: file: src/Native/libRevery_Native_stubs.a(shell_win32.o) has no symbols
          ocamlc packages/reason-skia/src/wrapped/bindings/.SkiaWrappedBindings.objs/byte/skiaWrappedBindings__Skia_generated_type_stubs.{cmi,cmo,cmt}
    File "packages/reason-skia/src/wrapped/bindings/skia_generated_type_stubs.ml", line 91, characters 4-63:
    91 |   | Struct ({ tag = "sk_fontmetrics_t"} as s'), "fMaxCharWidth" ->
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    Warning 11: this match case is unused.
    File "packages/reason-skia/src/wrapped/bindings/skia_generated_type_stubs.ml", line 94, characters 4-63:
    94 |   | Struct ({ tag = "sk_fontmetrics_t"} as s'), "fAvgCharWidth" ->
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    Warning 11: this match case is unused.
    File "packages/reason-skia/src/wrapped/bindings/skia_generated_type_stubs.ml", line 155, characters 4-78:
    155 |   | Ctypes_static.Primitive Cstubs_internals.Int64_t, "RGB_V_SK_PIXELGEOMETRY" ->
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    Warning 11: this match case is unused.
        ocamlopt packages/reason-skia/src/wrapped/bindings/.SkiaWrappedBindings.objs/native/skiaWrappedBindings__Skia_generated_type_stubs.{cmx,o}
    File "packages/reason-skia/src/wrapped/bindings/skia_generated_type_stubs.ml", line 91, characters 4-63:
    91 |   | Struct ({ tag = "sk_fontmetrics_t"} as s'), "fMaxCharWidth" ->
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    Warning 11: this match case is unused.
    File "packages/reason-skia/src/wrapped/bindings/skia_generated_type_stubs.ml", line 94, characters 4-63:
    94 |   | Struct ({ tag = "sk_fontmetrics_t"} as s'), "fAvgCharWidth" ->
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    Warning 11: this match case is unused.
    File "packages/reason-skia/src/wrapped/bindings/skia_generated_type_stubs.ml", line 155, characters 4-78:
    155 |   | Ctypes_static.Primitive Cstubs_internals.Int64_t, "RGB_V_SK_PIXELGEOMETRY" ->
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    Warning 11: this match case is unused.
             gcc packages/reason-sdl2/src/sdl2_wrapper.o (exit 1)
    (cd /Users/ragnarhardarson/.esy/3/b/revery-fc1e0d20/default/packages/reason-sdl2/src && /usr/bin/gcc -I /Users/ragnarhardarson/.esy/3________________________________________________________/i/esy_sdl2-2.0.14000-3a4931c9/include -I /Users/ragnarhardarson/.esy/source/i/revery__c93689c4/include -I /Users/ragnarhardarson/.esy/source/i/revery__c93689c4/src -x objective-c++ -g -I /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml -o sdl2_wrapper.o -c sdl2_wrapper.cpp)
    In file included from sdl2_wrapper.cpp:4:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/algorithm:639:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/initializer_list:46:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/cstddef:37:
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:1:1: error: expected unqualified-id
    4.09.1
    ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:3:3: error: invalid preprocessing directive
    # The version string is the first line of this file.
      ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:4:3: error: invalid preprocessing directive
    # It must be in the format described in stdlib/sys.mli
      ^
    In file included from sdl2_wrapper.cpp:4:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/algorithm:639:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/initializer_list:46:
    /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/cstddef:49:9: error: no member named 'ptrdiff_t' in the global namespace
    using ::ptrdiff_t;
          ~~^
    In file included from sdl2_wrapper.cpp:4:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/algorithm:640:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/type_traits:418:
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:1:1: error: expected unqualified-id
    4.09.1
    ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:3:3: error: invalid preprocessing directive
    # The version string is the first line of this file.
      ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:4:3: error: invalid preprocessing directive
    # It must be in the format described in stdlib/sys.mli
      ^
    In file included from sdl2_wrapper.cpp:4:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/algorithm:642:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/utility:205:
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:1:1: error: expected unqualified-id
    4.09.1
    ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:3:3: error: invalid preprocessing directive
    # The version string is the first line of this file.
      ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:4:3: error: invalid preprocessing directive
    # It must be in the format described in stdlib/sys.mli
      ^
    In file included from sdl2_wrapper.cpp:4:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/algorithm:643:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/memory:668:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/typeinfo:60:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/exception:83:
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:1:1: error: expected unqualified-id
    4.09.1
    ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:3:3: error: invalid preprocessing directive
    # The version string is the first line of this file.
      ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:4:3: error: invalid preprocessing directive
    # It must be in the format described in stdlib/sys.mli
      ^
    In file included from sdl2_wrapper.cpp:4:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/algorithm:643:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/memory:671:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/new:93:
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:1:1: error: expected unqualified-id
    4.09.1
    ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:3:3: error: invalid preprocessing directive
    # The version string is the first line of this file.
      ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:4:3: error: invalid preprocessing directive
    # It must be in the format described in stdlib/sys.mli
      ^
    In file included from sdl2_wrapper.cpp:4:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/algorithm:643:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/memory:673:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/limits:121:
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:1:1: error: expected unqualified-id
    4.09.1
    ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:3:3: error: invalid preprocessing directive
    # The version string is the first line of this file.
      ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:4:3: error: invalid preprocessing directive
    # It must be in the format described in stdlib/sys.mli
      ^
    fatal error: too many errors emitted, stopping now [-ferror-limit=]
    20 errors generated.
             gcc packages/reason-sdl2/src/stb_image.o (exit 1)
    (cd /Users/ragnarhardarson/.esy/3/b/revery-fc1e0d20/default/packages/reason-sdl2/src && /usr/bin/gcc -I /Users/ragnarhardarson/.esy/3________________________________________________________/i/esy_sdl2-2.0.14000-3a4931c9/include -I /Users/ragnarhardarson/.esy/source/i/revery__c93689c4/include -I /Users/ragnarhardarson/.esy/source/i/revery__c93689c4/src -x objective-c++ -g -I /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml -o stb_image.o -c stb_image.cpp)
    In file included from stb_image.cpp:2:
    In file included from /Users/ragnarhardarson/.esy/source/i/revery__c93689c4/include/stb_image.h:514:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/math.h:309:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/type_traits:417:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/cstddef:37:
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:1:1: error: expected unqualified-id
    4.09.1
    ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:3:3: error: invalid preprocessing directive
    # The version string is the first line of this file.
      ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:4:3: error: invalid preprocessing directive
    # It must be in the format described in stdlib/sys.mli
      ^
    In file included from stb_image.cpp:2:
    In file included from /Users/ragnarhardarson/.esy/source/i/revery__c93689c4/include/stb_image.h:514:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/math.h:309:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/type_traits:418:
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:1:1: error: expected unqualified-id
    4.09.1
    ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:3:3: error: invalid preprocessing directive
    # The version string is the first line of this file.
      ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:4:3: error: invalid preprocessing directive
    # It must be in the format described in stdlib/sys.mli
      ^
    In file included from stb_image.cpp:2:
    In file included from /Users/ragnarhardarson/.esy/source/i/revery__c93689c4/include/stb_image.h:514:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/math.h:310:
    In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/limits:121:
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:1:1: error: expected unqualified-id
    4.09.1
    ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:3:3: error: invalid preprocessing directive
    # The version string is the first line of this file.
      ^
    /Users/ragnarhardarson/.esy/3________________________________________________________/i/ocaml-4.9.1000-c0c8387c/lib/ocaml/version:4:3: error: invalid preprocessing directive
    # It must be in the format described in stdlib/sys.mli
      ^
    9 errors generated.
    error: command failed: 'dune' 'build' '-p' 'reason-harfbuzz,reason-skia,reason-sdl2,Revery' (exited with 1)
    esy-build-package: exiting with errors above...

  building revery@github:revery-ui/revery#d737fcd
esy: exiting due to errors above
```

</details>